lib/controls/ValueSelector.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+class ValueSelector extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    const {value, options, className, handleOnChange} = this.props;
+    const children = this.props.children;
+
+    if (children) {
+      return (
+        React.cloneElement(children,
```

Comment:
Ya, the fallbacks will be part of QueryBuilder. Its only when we invoke the ValueEditor where we pass in the fallback as a prop.

```jsx
const props = {
      operator, field,
      fallbackControl: (<Input type="text" />)
};

const editor = React.createElement(ValueEditor, props)
```

The ValueEditor can decide what to do in its `render`

```jsx
class MyEditor extends React.Component {
        render() {
            // render the custom editor or use this.props.fallbackControl as the return value
        }
}
```

HTH.